cluster.tf

```diff
@@ -127,6 +127,19 @@ resource "google_container_cluster" "primary" {
     daily_maintenance_window {
       start_time = var.maintenance_start_time
     }
+
+    dynamic "maintenance_exclusion" {
+      for_each = var.maintenance_exclusions
+      content {
+        exclusion_name = maintenance_exclusion.value.name
+        start_time     = maintenance_exclusion.value.start_time
+        end_time       = maintenance_exclusion.value.end_time
+
+        exclusion_options {
```

Comment:
Lets make this a dynamic block created iff maintenance_exclusion.value.exclusion_options is set.